### DISCUSSION TITLE
Commonly used Software integration

### DISCUSSION THREAD

**Discussion by Rakesh-123-cryp** [mesa/mesa#2077] [Ideas]
Softwares like SUMO and Comet are growing popular and mesa can include APIs to all these softwares to ease simulation and boost usage.

**Comment by tpike3** [comment]
@Rakesh-123-cryp Thanks for posting... could you provide links?

**Comment by Rakesh-123-cryp** [comment]
Sumo - https://github.com/eclipse-sumo/sumo

**Comment by Rakesh-123-cryp** [comment]
We can take the configuration file of the maps in SUMO and integrate it with our visualisation. Sumo generates real time traffic.

**Comment by EwoutH** [comment]
@Rakesh-123-cryp Coincidentally, I'm also into traffic modelling currently (on a mesoscopic level) for my master thesis. It would be interesting to have nice interactions with traffic modeling with mesa. I will be working on this in the next few weeks, and after that I probably have better ideas on how Mesa could integrate with external software.

If you're also interested in macroscopic or mesoscopic network traffic flow simulation, you should check out [UXsim](https://github.com/toruseo/UXsim). It does exactly that but then in pure Python.
